tests/Unit/Evolution/Systems/Burgers/Test_Observe.cpp

```diff
@@ -0,0 +1,272 @@
+    CHECK(reduction_data.pack_size() == 3);
+    results.time = std::get<0>(reduction_data.data());
+    results.number_of_grid_points = std::get<1>(reduction_data.data());
+    results.u_error = std::get<2>(reduction_data.data());
+  }
+};
+
+MockContributeReductionData::Results MockContributeReductionData::results{};
+
+struct Metavariables;
+
+struct ElementComponent {
+  using component_being_mocked = void;
+
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<1>;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using initial_databox = db::compute_databox_type<db::AddSimpleTags<>>;
+  using action_list = tmpl::list<>;
+};
+
+struct MockObserverComponent {
+  using component_being_mocked = observers::Observer<Metavariables>;
+  using replace_these_simple_actions =
+      tmpl::list<observers::Actions::ContributeVolumeData,
+                 observers::Actions::ContributeReductionData>;
+  using with_these_simple_actions =
+      tmpl::list<MockContributeVolumeData, MockContributeReductionData>;
+
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = int;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using initial_databox = db::compute_databox_type<db::AddSimpleTags<>>;
+  using action_list = tmpl::list<>;
+};
+
+struct Metavariables {
+  using system = Burgers::System;
+  using component_list = tmpl::list<ElementComponent, MockObserverComponent>;
+  using const_global_cache_tag_list =
+      tmpl::list<OptionTags::AnalyticSolution<Burgers::Solutions::Linear>>;
+};
+
+template <typename ObserveEvent>
+void test_observe(const ObserveEvent& observe) noexcept {
+  const Burgers::Solutions::Linear analytic_solution(0.0);
+  const ElementComponent::array_index array_index(ElementId<1>(2));
+  const std::string element_name =
+      get_output(static_cast<ElementId<1>>(array_index));
+  const Mesh<1> mesh(5, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto);
+  const double observation_time = 2.0;
+  const tnsr::I<DataVector, 1> coords{{{{4., 3., 2., 1., 0.}}}};
+  const Scalar<DataVector> u{{{{1., 2., 3., 4., 5.}}}};
+  const Scalar<DataVector> u_analytic =
+      tuples::get<Burgers::Tags::U>(analytic_solution.variables(
+          coords, observation_time, tmpl::list<Burgers::Tags::U>{}));
+  const Scalar<DataVector> u_error{get(u) - get(u_analytic)};
+
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+
+  MockRuntimeSystem::TupleOfMockDistributedObjects dist_objects{};
+  tuples::get<typename MockRuntimeSystem::template MockDistributedObjectsTag<
+      ElementComponent>>(dist_objects)
+      .emplace(array_index,
+               ActionTesting::MockDistributedObject<ElementComponent>{});
+  tuples::get<typename MockRuntimeSystem::template MockDistributedObjectsTag<
+      MockObserverComponent>>(dist_objects)
+      .emplace(0,
```

Comment:
I'd prefer to replace the literal `0` with some variable name that reminds the reader why it is there, e.g. `fake_singleton_id`

Reply:
we agree with @wthrowe